src/main/java/com/google/cloud/tools/appengine/cloudsdk/CloudSdk.java

```diff
@@ -231,36 +248,39 @@ public Path getJarPath(String jarName) {
   }
 
   /**
-   * Checks whether the Cloud SDK Path with necessary components is installed in
-   * the expected location.
+   * Checks whether the Cloud SDK Path with is valid.
    *
-   * @throws AppEngineJavaComponentsNotInstalledException when the App Engine Java 
-   *     components are not installed in the Cloud SDK
-   * @throws AppEngineException when the Cloud SDK is not installed where expected
+   * @throws CloudSdkNotFoundException when the Cloud SDK is not installed where expected
    */
-  public void validate() throws AppEngineException {
+  public void validateCloudSdk() throws CloudSdkNotFoundException {
     if (sdkPath == null) {
```

Comment:
That's not useful for the IDE use-cases. We want to present actionable
errors to the user and we don't want to pass through your message strings
since we support localization.

On Mon, Sep 12, 2016 at 3:47 PM, meltsufin <notifications@github.com> wrote:

> In src/main/java/com/google/cloud/tools/appengine/cloudsdk/CloudSdk.java
> <https://github.com/GoogleCloudPlatform/appengine-plugins-core/pull/209#discussion_r78441437>
> :
>
> >     */
> > -  public void validate() throws AppEngineException {
> > +  public void validateCloudSdk() throws CloudSdkNotFoundException {
> >      if (sdkPath == null) {
>
> I think for now we just want to let clients know if the Cloud SDK is valid
> or not. Details are in the exception message.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/GoogleCloudPlatform/appengine-plugins-core/pull/209/files/b4db3945ca1769ffd1bde02f454625c91b3d3e57#r78441437>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AHf5HRs3SaxIHx4R2a2Mg4h-uVkym_14ks5qpaxrgaJpZM4J68gI>
> .
>